### ISSUE TITLE
Add/fix documentation around VideoMAEForPretraining's `bool_masked_pos` argument

### ISSUE THREAD

**Issue by nateraw** [huggingface/transformers#21484]
### System Info

N/A

### Who can help?

@NielsRogge @amyeroberts 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The `VideoMAEForPretraining` class is missing a docstring for an important argument to the `forward` function, `bool_masked_pos`. Can see this in the docs for the `main` branch [here](https://huggingface.co/docs/transformers/main/en/model_doc/videomae#transformers.VideoMAEForPreTraining.forward)...it's listed in the function signature but not in the documented arguments.

There is an [example code snippet](https://huggingface.co/docs/transformers/main/en/model_doc/videomae#transformers.VideoMAEForPreTraining.forward.example) that creates this input tensor that you can see in the docs, but I'm not sure it's correct when applied to video inputs with `batch_size > 1` (if not done carefully). When I calculated it example-by-example, I was getting errors, which is what led me to this issue.

I dug a little deeper and noticed that the test suite for VideoMAE has [different logic](https://github.com/huggingface/transformers/blob/5b49376202863d3798d2ff8a8ba61590542a1141/tests/models/videomae/test_modeling_videomae.py#L145-L149) for creating `bool_masked_pos`. When I updated my training code to use that logic, my problems went away. I assume this is related to the note in the tests that mentions each video needing the same number of masked patches.

### Expected behavior

To resolve this issue, I think we should:

1. Add a docstring for `bool_masked_pos` that explains what it is for users
2. Update the example snippet to use the correct logic from the test suite.

Even better I guess would be to handle this for users in the `ImageProcessor`, but I'll leave that for a separate issue.

**Comment by NielsRogge** [comment]
Thanks for raising this issue!

VideoMAE indeed uses the same mask ratio (number of masked patches) per video to make batching possible. See [this class](https://github.com/MCG-NJU/VideoMAE/blob/main/masking_generator.py) which the authors use to generate boolean masks.

In the tests, I just define the the same mask for all examples in the batch, but in practice one would just different masks (but with the same mask ratio) in a batch.

**Comment by nateraw** [comment]
@NielsRogge can you add the docstring for the param, please? :) I think you'd be best person to write it.

As for fixing the example...maybe we write a quick function and include it in the snippet.

**Comment by amyeroberts** [comment]
Thanks for raising @nateraw ! We should definitely update the docstring and example snippet.

I'm not sure `bool_mask_pos` should be generated in the image processor. The image processor takes the images and makes them ready to be passed into the model, however it doesn't handle other transformations which might be part of the training procedure e.g. augmentation. For similar tasks, where input tokens are randomly masked, this in handled in e.g. `DataCollatorForLanguageModeling`, rather than the tokenizer.

**Comment by NielsRogge** [comment]
Reopening as it's not resolved yet
